File: Programmers/Level2/카펫/카펫.js
function solution(brown, yellow) {
    let total = brown + yellow;
    for (let i = 3; i < parseInt(total ** 0.5) + 1; i++) {
        if (total % i === 0 && 2 * parseInt(total / i) + 2 * i - 4 == brown) {
            return [parseInt(total / i), i];
        }
    }
}

File: Programmers/Level2/카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    
    for b in range(1, total + 1):
        if not total % b:
            a = total / b
            if a >= b:
                if (2 * a) + (2 * b) - 4 == brown:
                    return [a, b]